pkg/vulnerabilityreport/io.go

```diff
@@ -93,9 +96,68 @@ func (r *readWriter) FindByOwner(ctx context.Context, owner kube.Object) ([]v1al
 	if err != nil {
 		return nil, err
 	}
+	// no reports found for provided owner, look for reports in related replicaset
+	if len(list.Items) == 0 && (owner.Kind == kube.KindDeployment || owner.Kind == kube.KindPod) {
+		rsName, found := r.findRelatedReplicaset(ctx, owner)
+		if !found {
+			return nil, nil
+		}
+
+		list, err = r.starboardClientset.AquasecurityV1alpha1().VulnerabilityReports(owner.Namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.Set{
+				kube.LabelResourceKind:      string(kube.KindReplicaSet),
+				kube.LabelResourceName:      rsName,
+				kube.LabelResourceNamespace: owner.Namespace,
+			}.String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return list.DeepCopy().Items, nil
 }
 
+// findRelatedReplicaset attempts to find the replicaset that is associated with
+// the given owner. If the owner is a Deployment, it will look for a replicaset
+// that is controlled by the Deployment. If the owner is a Pod, it will look for
+// the replicaset that owns the Pod.
+func (r *readWriter) findRelatedReplicaset(ctx context.Context, owner kube.Object) (rsName string, found bool) {
+	switch owner.Kind {
+	// find a replicaset controlled by the deployment
+	case kube.KindDeployment:
+		rsList, err := r.kubernetesClientset.AppsV1().ReplicaSets(owner.Namespace).List(ctx, metav1.ListOptions{})
```

Comment:
I know that this would require loading Deployment object, but maybe we can use label selector (deployment.Spec.Selector) to retrieve ReplicaSets instead of loading all of them in the specified namespace.

Reply:
I think that's a great idea, likely more efficient if there's a very large number of ReplicaSets.